TestClasses/TestStageChangeDomain.cls

```diff
@@ -0,0 +1,20 @@
+@isTest
+public class TestStageChangeDomain 
+{
+	
+    static testmethod void testStageChangeDomain()
+    {
+        Opportunity testOpp = new Opportunity();
+        testOpp.Name = 'testOppName';
+        testOpp.StageName = 'Prospecting';
+        testOpp.CloseDate = System.today().addDays(-2);
+        insert testOpp;
+        
+        testOpp.StageName = 'Closed Won';
+        update testOpp;
+        List<Opportunity> list1 = [select CloseDate , StageName from Opportunity where id =: testOpp.id];
+        System.assertEquals('Closed Won', list1.get(0).StageName);
+        System.assertEquals(System.today(), list1.get(0).CloseDate);
```

Comment:
verify the assertion.

Reply:
Sir, it is Today's Date as when we change the Stage the trigger change the date to today's date